modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts

```diff
@@ -0,0 +1,126 @@
+import { AddEventsBehaviour, AlloyComponent, AlloyEvents, Behaviour, Button, Focusing, Input, Keying, NativeEvents, Representing, SketchSpec } from '@ephox/alloy';
+import { Cell, Fun, Id, Optional } from '@ephox/katamari';
+import { Focus, SugarElement, Traverse } from '@ephox/sugar';
+
```

Comment:
So as a high-level overview of what you are doing here

* spec.updateText is used to update the input field to the appropriate value
* the DOM structure is a wrapper containing [minus button, input, plus button]
* pressing Escape on the wrapper isn't handled, and should be handled by the calling code
* pressing Escape within the wrapper should focus the wrapper
* pressing Enter on the wrapper should focus the 1st child inside the wrapper
* pressing Enter within the wrapper should do nothing, and isn't handled
* you use cells to keep values so that you can refer to them across components
* when plus or minus is triggered, you identify the unit via regex, and try to add / subtract and appropriate step size for that unit based on a config
* if the value is larger than 0, you keep it ... otherwise you change to the parsed value before adding/subtracting
* on clicking +/-, you execute the action to make the appropriate formatting change in the editor
* on typing in the input, you execute the action to make the appropriate formatting change in the editor. 
* on typing in the input, you do not validate the data

Does that sound right?